### PR TITLE
Move to Scala 2.13.0-M4

### PR DESCRIPTION
**Pull request by SethTisue**
this isn't mergeable.  I only did the obvious build changes, but
code changes are needed in order to adapt to the 2.13 collections

putting this out there hoping someone else will run with it.

references #167

### CODE REVIEW

**Comment by SethTisue** [comment]
the compilation errors on 2.13 can be seen by building this PR, but they are also seen at https://scala-ci.typesafe.com/job/scala-2.13.x-integrate-community-build/1226/consoleText

**Comment by xuwei-k** [comment]
almost done https://github.com/xuwei-k/play-json/commit/13785d926514a1447ada5ba60df5084758f6f7b0

but there are same `Map` ordering issues as another json libraries 😕 

```
[info]   - when to generated Writes *** FAILED ***
[info]     {"id":"foo","ls":[1.2,23.45],"set":[1,7,3,4],"seq":[{"_1":2,"_2":"bar"},{"_1":4,"_2":"lorem"},{"_1":5,"_2":"ipsum"}],"scores":{"A1":0.10000000149011612,"EF":12.300000190734863}} did not equal {"id":"foo","ls":[1.2,23.45],"set":[1,3,4,7],"seq":[{"_1":2,"_2":"bar"},{"_1":4,"_2":"lorem"},{"_2":"ipsum","_1":5}],"scores":{"A1":0.10000000149011612,"EF":12.300000190734863}} (MacroSpec.scala:496)
[info]   - when to generate Format *** FAILED ***
[info]     {"id":"foo","ls":[1.2,23.45],"set":[1,7,3,4],"seq":[{"_1":2,"_2":"bar"},{"_1":4,"_2":"lorem"},{"_1":5,"_2":"ipsum"}],"scores":{"A1":0.10000000149011612,"EF":12.300000190734863}} did not equal {"id":"foo","ls":[1.2,23.45],"set":[1,3,4,7],"seq":[{"_1":2,"_2":"bar"},{"_1":4,"_2":"lorem"},{"_2":"ipsum","_1":5}],"scores":{"A1":0.10000000149011612,"EF":12.300000190734863}} (MacroSpec.scala:496)
```

- https://github.com/spray/spray-json/pull/263/files#r200099063
- https://github.com/argonaut-io/argonaut/pull/308/files#diff-853736425f65f2850fb9d448e006b1d8L153
- https://github.com/json4s/json4s/commit/965233702520bb2f53418a6dd860de78bcdd4a69#diff-566fb4b6f211045a153ea0210a837da9R83

**Comment by marcospereira** [comment]
Thanks, @SethTisue and @xuwei-k.

@xuwei-k I've cherry-picked your commit here and added two more on top of it.

@gmethvin you may want to review this.

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
-- 2.12.4
+- 2.13.0-M4
+- 2.12.6
 - 2.11.12
-- 2.10.7
 jdk:
 - oraclejdk8
 - oraclejdk9
 script:
-- sbt -DscalaJSStage=full ++$TRAVIS_SCALA_VERSION validateCode mimaReportBinaryIssues testOnly
```
Comment (marcospereira) [inline .travis.yml]: We should either start to release play-json 2.7.0 and filter the binary compatibility issues, or we will need to have `scala-2.13.0-M4` to handle the necessary changes.

WDYT?

Comment (SethTisue) [reply .travis.yml]: if I understand the problem correctly, you could do `mimaPreviousVersion := (CrossVersion.partialVersion(scalaVersion.value) match { case Some((2, n)) if n <= 12 => Some(...); case _ => None } // TODO update once Scala 2.13.0 final is released`, that kind of thing? if `mimaPreviousVersion` is `None` then `mimaReportBinaryIssues` prints a warning but doesn't fail the build.

Comment (marcospereira) [reply .travis.yml]: Noticed this is already handled here:

https://github.com/playframework/play-json/blob/80f42eab24f0c14f6569456d8dd602b9367e6e98/build.sbt#L47-L54

So nothing to do besides add mima filters and add the task back in Travis config.

**Comment by marcospereira** [comment]
> what's the next step here?

I guess we can merge and do a milestone release (2.7.0) since we are breaking binary compatibility. Will merge it now and do a new release later.

**Comment by SethTisue** [comment]
does this help with the bincompat issue around `Factory` vs `CanBuildFrom`? https://github.com/scala/scala-collection-compat/pull/115

**Pull request merged**